src/test/java/org/jenkinsci/plugins/durabletask/WindowsBatchScriptTest.java

```diff
@@ -167,13 +169,29 @@ private void testWithPath(String path) throws Exception {
             "echo zwei\n" +
             "goto :eins\n";
         Controller c = new WindowsBatchScript(script).launch(new EnvVars(), ws, launcher, listener);
-        while (c.exitStatus(ws, launcher, listener) == null) {
-            Thread.sleep(100);
-        }
+        awaitCompletion(c);
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        c.writeLog(ws, baos);
+        assertEquals(0, c.exitStatus(ws, launcher, listener).intValue());
+        c.cleanup(ws);
+    }
+
+    @Test public void unicodeChars() throws Exception {
+        Controller c = new WindowsBatchScript("echo Helló, Wõrld ®").launch(new EnvVars(), ws, launcher, listener);
+        awaitCompletion(c);
         ByteArrayOutputStream baos = new ByteArrayOutputStream();
         c.writeLog(ws, baos);
         assertEquals(0, c.exitStatus(ws, launcher, listener).intValue());
+        String log = baos.toString("UTF-8");
+        assertTrue(log, log.contains("Helló, Wõrld ®"));
         c.cleanup(ws);
     }
 
+    private void awaitCompletion(Controller c) throws IOException, InterruptedException {
+        while (c.exitStatus(ws, launcher, listener) == null) {
+            Thread.sleep(100);
+        }
+        Thread.sleep(3000); // Pause allows slower systems to exit binary before cleanup is attempted
```

Comment:
This is kind of a long sleep, do things flake if this is reduced to one second or less? Or could you directly check for the process to exit or attempt to delete a file used by the binary in a loop instead and exit once it succeeds?

Reply:
So I know there is a retry delete attempt somewhere in either the test package or FilePath and that was looping too fast. I tried getting the sleep down to 1.5 secs, but that's still not working. I think I will need to directly check the process or add in some other retry in cleanup

Reply:
So there is[ a lot of tuning](https://github.com/jenkinsci/jenkins/blob/2f612fcac9194419e4a73bc2ee5b7ef422b9d6dd/core/src/main/java/hudson/Util.java#L1767) that you can do in `hudson.Util` to handle the `deleteRecursive` that's failing. The only problem is that they are all `@Restricted(value = NoExternalUse.class)`.

Reply:
Okay, added additional retry logic in `FileMonitoringTask.cleanup()`. Was able to remove sleep in favor of this behavior.

see d7d4517

Reply:
sigh...looks like the tuning worked locally, but does not work for CI?